tests/Integration/Behaviour/Features/Context/Domain/Product/ProductImageFeatureContext.php

```diff
@@ -310,11 +323,35 @@ private function parseGeneratedImagePath(int $imageId, string $imageTypeName): s
     public function assertProductHasNoImages(string $productReference): void
     {
         Assert::assertEmpty(
-            $this->getProductImages($productReference),
+            $this->getProductImages($productReference, ShopConstraint::shop($this->getDefaultShopId())),
             sprintf('No images expected for product "%s"', $productReference)
         );
     }
 
+    /**
+     * @Then product :productReference should have no images for shop :shops
+     *
+     * @param string $productReference
+     * @param string $shopReferences
+     */
+    public function assertProductHasNoImagesForShops(string $productReference, string $shopReferences): void
+    {
+        //todo: use transformer
+        $shopIds = array_map(
+          function (string $shopReference): int {
+              return $this->getSharedStorage()->get(trim($shopReference));
+          },
+            explode(',', $shopReferences)
+        );
```

Comment:
```suggestion
        $shopIds = $this->referencesToIds($shopReferences);
```
We have this useful helper method now